eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -815,15 +845,16 @@ showValue v = show c ++ "\n" ++ unlines (map trans ts)
 -- | Turn the nested 'MaryValue' map-of-maps representation into a flat sequence
 -- of policyID, asset name and quantity, plus separately the ada quantity.
 gettriples' :: MaryValue crypto -> (Integer, [(PolicyID crypto, AssetName, Integer)], [PolicyID crypto])
-gettriples' (MaryValue c m1) = (c, triples, bad)
+gettriples' v@(MaryValue _ (MultiAsset m)) = (c, triples, Map.keys (Map.filter Map.null m))
```

Comment:
I suggest we remove this function, this is only used in pretty printing. What we actually need for pretty printer is:
```haskell
-- | If MutliAsset is a malformed value, not in canonical form, return the violators:
getBadMultiAsset (MultiAsset m) = Map.keys (Map.filter Map.null m)
```

See my next comment in `libs/cardano-ledger-pretty/src/Cardano/Ledger/Pretty/Mary.hs` why this works

Reply:
I would write this new function where it's needed in pretty and remove `gettriples'`  from here completely, what do you think?

Reply:
And actually, the whole `gettriples` function: 

```
gettriples :: MaryValue crypto -> (Integer, [(PolicyID crypto, AssetName, Integer)])
gettriples (MaryValue c ma) = (c, flattenMultiAsset ma)
```

doesn't seem to bring a lot of value- I would remove it completely and instead expose only `flattenMultiAsset`

Reply:
Actually nevermind, I see that this `gettriples` doesn't get exposed from the module, I will leave it alone